src/qcodes_contrib_drivers/drivers/Lakeshore/M81_SSM.py

```diff
@@ -0,0 +1,171 @@
+            full_source_list.append((f"{i} : {self.ask(f'SOURce{i}:MODel?')}"))
+        return full_source_list
+
+    def _get_sense_module_list(self):
+        full_sense_list = []
+        for i in range (1, self.n_chan+1):
+            full_sense_list.append((f"{i} : {self.ask(f'SENSe{i}:MODel?')}"))
+        return full_sense_list
+
+    data_source_types = {
+        'RTIMe': float,
+        'SAMPlitude': float,
+        'SOFFset': float,
+        'SFRequency': float,
+        'SRANge': float,
+        'SVLimit': lambda s: bool(int(s)),
+        'SILimit': lambda s: bool(int(s)),
+        'MDC': float,
+        'MRMs': float,
+        'MPPeak': float,
+        'MNPeak': float,
+        'MPTPeak': float,
+        'MX': float,
+        'MY': float,
+        'MR': float,
+        'MTHeta': float,
+        'MRANge': float,
+        'MOVerload': lambda s: bool(int(s)),
+        'MSETtling': lambda s: bool(int(s)),
+        'MUNLock': lambda s: bool(int(s)),
+        'MRFRequency': float,
+        'GPIStates': int,
+        'GPOStates': int,
+    }
+    
+    def _configure_stream_elements(self, data_sources):
+        """
+        Sets the elements to include in the data stream. Takes a list of pairs of data source mnemonic and channel number. Up to 10 pairs.
+        """
+        elements = ','.join('{},{}'.format(mnemonic, index) for (mnemonic, index) in data_sources)
+        self.write('TRACe:FORMat:ELEMents {}'.format(elements))
+        
+    def stream_data(self, rate, num_points, *data_sources, transpose_data=True):
+        """Generator object to stream data from the instrument.
+
+            Args:
+                rate (int): Desired transfer rate in samples/sec. The maximum stream rate is 5000 samples/s.
+                num_points (int): Number of points to return. None to stream indefinitely.
+                data_sources (str, int): Variable length list of pairs of (DATASOURCE_MNEMONIC, CHANNEL_INDEX).
+                transpose_data (bool): transposes the data retured to get an array for each parameter streamed.
+
+            Yields:
+                Stream data as a list
+        """
+
+        self._configure_stream_elements(data_sources)
+
+        self.write('TRACe:RESEt')
+
+        self.write('TRACe:FORMat:ENCOding B64')
+
+        self.write('TRACe:RATE {}'.format(rate))
+
+        #start streaming
+        self.write('TRACe:STARt {}'.format(num_points))
+        print('Streaming...')
+
+        # data counter
+        count = 0
+        while count < num_points:
+            time.sleep(1) # or what ever small value
+            count = int(self.ask("TRACe:DATA:COUNt?"))
+            print(f"\rBuffered points: {count}         ", end="", flush=True)
+    
+        time.sleep(1)
+        print('\r')
+        
+        # pull data
+        string_bytes = base64.b64decode(self.ask('TRACe:DATA:ALL?')).hex()
+
+        data_format = f"<{self.ask('TRACe:FORMat:ENCOding:B64:BFORmat?').strip('\"')}"
```

Comment:
One way to work around the issue in python 3.11 is to extract this command out into a value

```suggestion
        format_val = self.ask('TRACe:FORMat:ENCOding:B64:BFORmat?').strip('\"')
        data_format = f"<{format_val}"
```